engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java

```diff
@@ -107,11 +104,44 @@ public <T> Optional<T> addIndexedEmbeddedIfIncluded(
 		}
 	}
 
+	private boolean shouldNotBeExcluded(IndexedEmbeddedDefinition definition) {
+		String prefixToParse = unconsumedPrefix + definition.relativePrefix();
+		int afterPreviousDotIndex = 0;
+		int nextDotIndex = prefixToParse.indexOf( '.', afterPreviousDotIndex );
+		boolean shouldInclude = true;
+		while ( nextDotIndex >= 0 ) {
+			// Make sure to mark the paths as encountered in the filter
+			String objectNameRelativeToFilter = prefixToParse.substring( 0, nextDotIndex );
+
+			//TODO: to Yoann: I am not convinced if this is a correct way to address it or not, but based on the tests
+			// we have for prefixes I've guessed that if we have a prefixed embedded then `unconsumedPrefix` will have that prefix in it
+			// and we not necessarily need to "include" a path-with-prefix in the filter if we want to have a property to be included ( DeepPathWithLeadingPrefixCase )
```

Comment:
These 3 would fail:

TestInvalidPaths.testDepthMatchesPathMarkedAsEncountered:
```
org.hibernate.search.util.common.SearchException: HSEARCH000520: Hibernate Search encountered failures during bootstrap. Failures:

    Hibernate ORM mapping: 
        type 'org.hibernate.search.test.embedded.path.validation.A': 
            failures: 
              - HSEARCH000216: An @IndexedEmbedded defines includePaths filters that do not match anything. Non-matching includePaths filters: [c.indexed]. Encountered field paths: [prefixedc]. Check the filters for typos, or remove them if they are not useful.
```

TestInvalidPaths.testDepthExceedsPathMarkedAsEncountered:
```
org.hibernate.search.util.common.SearchException: HSEARCH000520: Hibernate Search encountered failures during bootstrap. Failures:

    Hibernate ORM mapping: 
        type 'org.hibernate.search.test.embedded.path.validation.A': 
            failures: 
              - HSEARCH000216: An @IndexedEmbedded defines includePaths filters that do not match anything. Non-matching includePaths filters: [c.indexed]. Encountered field paths: [prefixedc]. Check the filters for typos, or remove them if they are not useful.
```

TestInvalidPaths.testInvalidDeepSimplePathWithLeadingPrefix:
```
org.hibernate.search.util.common.SearchException: HSEARCH000520: Hibernate Search encountered failures during bootstrap. Failures:

    Hibernate ORM mapping: 
        type 'org.hibernate.search.test.embedded.path.validation.A': 
            failures: 
              - HSEARCH000216: An @IndexedEmbedded defines includePaths filters that do not match anything. Non-matching includePaths filters: [c.indexed]. Encountered field paths: [prefixedc]. Check the filters for typos, or remove them if they are not useful.
        type 'org.hibernate.search.test.embedded.path.validation.DeepPathWithLeadingPrefixCase': 
            failures: 
              - HSEARCH000216: An @IndexedEmbedded defines includePaths filters that do not match anything. Non-matching includePaths filters: [b.c.indexed, b.c.dne]. Encountered field paths: [notJustAb]. Check the filters for typos, or remove them if they are not useful.
```

Before any changes - we'd check that some paths are included and be happy about it. It wouldn't matter that the path that we get at that point is "not included" when the filter is called inside the while loop. And now that we have to make this additional check, we'd need to have a path that matches something 🥲

Reply:
Can you push the code that leads to these errors somewhere, so that I have a closer look?

Reply:
yup, sure, 🙈  pushed it here - https://github.com/marko-bekhta/hibernate-search/blob/feat/HSEARCH-1182-exclude-paths-indexed-embedded-error-reproducer and added a [comment with failing tests](https://github.com/marko-bekhta/hibernate-search/blob/e9e2ee4009c547798d1e0e854b0d750c45288cb2/engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java#L94C1-L103)

Reply:
You actually discovered a bug, and it was hidden in plain sight in one of our tests...

I sent a PR to your branch: https://github.com/marko-bekhta/hibernate-search/pull/135